lib/src/router/normalized_path.dart

```diff
@@ -9,6 +11,7 @@ import 'lru_cache.dart';
 ///
 /// Instances are interned using an LRU cache for efficiency, meaning identical
 /// normalized paths will often share the same object instance.
+@immutable
 class NormalizedPath {
   /// Cache of interned instances
   static var interned = LruCache<String, NormalizedPath>(10000);
```

Comment:
Q: Will this class also be used for incoming requests (as I didn't find any current usage outside of the router)?

Just wondering where the "duplicate" instances would come from that we need _internal_ caching.

Reply:
>  It is quite common for a server to route the same paths repeatedly.

Yeah, I was wondering whether I missed that (that the normalized path is also used in incoming requests), and not just when building the router tree.